providers/base/bin/prime_offload_tester.py

```diff
@@ -0,0 +1,357 @@
+# as published by the Free Software Foundation.
+#
+# Checkbox is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Checkbox.  If not, see <http://www.gnu.org/licenses/>.
+
+import sys
+import threading
+import subprocess
+import time
+import re
+import json
+import argparse
+import logging
+import os
+
+
+class PrimeOffloader:
+    """
+    A class used to execute process to specific GPU.
+    Have to run this as root.
+
+    :attr logger: console logger
+    :type logger: obj
+
+    :attr check_result:
+        store the result of checking offloading is ok or not.
+    :type check_result: int
+    """
+
+    logger = logging.getLogger()
+    check_result = False
+
+    def find_card_id(self, pci_name):
+        """
+        use pci name to find card id under /sys/kernel/debug/dri
+
+        :param pci_name: pci device name in NNNN:NN:NN.N format
+        :type card_name: str
+
+        : returns : card id
+        : rtype : str
+        """
+        if not re.match("[0-9]{4}:[0-9]{2}:[0-9]{2}.[0-9]", pci_name):
+            raise RuntimeError("pci name format error")
+
+        try:
+            cmd = ["grep",
+                   "-lr",
+                   "--include=name",
+                   pci_name,
+                   "/sys/kernel/debug/dri"]
+
+            card_path = subprocess.check_output(cmd,
+                                                universal_newlines=True)
+            return card_path.split('/')[5]
+        except (IndexError, AttributeError) as e:
+            raise RuntimeError("return value format error {}".format(e))
+        except subprocess.CalledProcessError as e:
+            raise RuntimeError(e)
+
+    def find_card_name(self, pci_name):
+        """
+        use pci name to find card name by lshw
+
+        :param pci_name: pci device name in NNNN:NN:NN.N format
+        :type card_name: str
+
+        : returns : card name
+        : rtype : str
+        """
+        cmd = ["lshw", "-c", "display", "-json"]
+        try:
+            card_infos = subprocess.check_output(cmd,
+                                                 universal_newlines=True)
+            infos = json.loads(card_infos)
+            for info in infos:
+                if pci_name in info['businfo']:
+                    return info['product']
+            raise RuntimeError("Card name not found")
+        except (KeyError, TypeError, json.decoder.JSONDecodeError) as e:
+            raise RuntimeError("return value format error {}".format(e))
+        except subprocess.CalledProcessError as e:
+            raise RuntimeError(e)
+
+    def check_for_kernel_bug(self, b, a):
+        """
+        for 6.5 kernel, a bug of dir debugfs should be fixed.
```

Comment:
This should state what the function does.
From what I can tell by looking a the code it checks if the second string is longer by one line and checks whether the "after" one has "Xorg" or "Xwayland" in the last line.